src/components/Article.jsx

```diff
@@ -9,20 +9,25 @@ import formatDate from "../utils/formatDate";
 const Article = () => {
     const {article_id} = useParams();
     const {articleData, isLoading} = useArticle(article_id);
-    const [hasVoted, setHasVoted] = useState(false);
+    const [hasVoted, setHasVoted] = useState(Boolean(localStorage.getItem(`voted${article_id}`)));
+    const [error, setError] = useState(null);
 
     const upvotehandler = () => {
         const vote = {inc_votes: 1};
         articleData.votes += 1;
-        setHasVoted(true);
-        patchArticleById(article_id, vote);
+        localStorage.setItem(`voted${article_id}`, true);
+        setHasVoted(Boolean(localStorage.getItem(`voted${article_id}`)));
+        patchArticleById(article_id, vote)
+        .catch(err => setError(err));
     }
 
     const downvoteHandler = () => {
         const vote = {inc_votes: -1};
         articleData.votes -= 1;
-        setHasVoted(true);
-        patchArticleById(article_id, vote);
+        localStorage.setItem(`voted${article_id}`, true);
+        setHasVoted(Boolean(localStorage.getItem(`voted${article_id}`)));
+        patchArticleById(article_id, vote)
+        .catch(err => setError(err));
```

Comment:
Same here - we lowered the vote by one but this could be changed back in the .catch to illustrate the old vote again